pkg/types/vsphere/platform.go

```diff
@@ -4,25 +4,41 @@ package vsphere
 type Platform struct {
 	// VCenter is the domain name or IP address of the vCenter.
 	VCenter string `json:"vCenter"`
+
 	// Username is the name of the user to use to connect to the vCenter.
 	Username string `json:"username"`
+
 	// Password is the password for the user to use to connect to the vCenter.
 	Password string `json:"password"`
+
 	// Datacenter is the name of the datacenter to use in the vCenter.
 	Datacenter string `json:"datacenter"`
+
 	// DefaultDatastore is the default datastore to use for provisioning volumes.
 	DefaultDatastore string `json:"defaultDatastore"`
+
 	// Folder is the name of the folder that will be used and/or created for
 	// virtual machines.
 	Folder string `json:"folder,omitempty"`
+
 	// Cluster is the name of the cluster virtual machines will be cloned into.
 	Cluster string `json:"cluster,omitempty"`
+
 	// ClusterOSImage overrides the url provided in rhcos.json to download the RHCOS OVA
 	ClusterOSImage string `json:"clusterOSImage,omitempty"`
+
 	// APIVIP is the virtual IP address for the api endpoint
 	APIVIP string `json:"apiVIP,omitempty"`
+
 	// IngressVIP is the virtual IP address for ingress
 	IngressVIP string `json:"ingressVIP,omitempty"`
+
 	// DNSVIP is the virtual IP address for DNS
 	DNSVIP string `json:"dnsVIP,omitempty"`
+
+	// DefaultMachinePlatform is the default configuration used when
+	// installing on VSphere for machine pools which do not define their own
+	// platform configuration.
+	// +optional
+	DefaultMachinePlatform *MachinePool `json:"defaultMachinePlatform,omitempty"`
```

Comment:
@patrickdillon can you add
```
Network string `json:"network,omitempty"`
```
 since I forgot it.

Reply:
@jcpowermac I can add `Network` to the `Platform`. `omitempty` means it is optional, but it looks like the provider requires a NetworkName: https://github.com/openshift/machine-api-operator/blob/master/pkg/apis/vsphereprovider/v1alpha1/vsphereproviderconfig_types.go#L32 WDYT?

Reply:
@patrickdillon UPI doesn't need it and if we don't include `omitempty` then I am assuming it would become required parameter for UPI?

Reply:
I am trying to gain some background on this, but have to run to another meeting in 5 mins. 

> UPI doesn't need it
But we do need to pass a value to terraform, right? https://github.com/openshift/installer/pull/2841/files#diff-779e91f6248cdf8645f11e13f8849d28R47

I think we just need to make sure those values match here.

Reply:
@patrickdillon yeah for IPI terraform and the machine-api-operator:
https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/reconciler.go#L324-L325